pkg/egressgateway/manager.go

```diff
@@ -263,6 +259,10 @@ func newEgressGatewayManager(p Params) (*Manager, error) {
 			if probes.HaveLargeInstructionLimit() != nil {
 				return fmt.Errorf("egress gateway needs kernel 5.2 or newer")
 			}
+			manager.endpointStore, err = manager.endpoints.Store(hc)
```

Comment:
I think this will block until the store is initialized, which is probably not what we want?

Reply:
Yep this will block the starting and potentially will fail the start if this takes too long (the default start timeout is 5 minutes). Move this into `processEvents`.

Reply:
> Move this into processEvents

makes sense, thanks!

Reply:
Those uses of `Store()` in a `Start` hook are definitely dubious. Looks like bgpv1 does it on the `CiliumBGPPeeringPolicy`, which is slightly safer in the sense that there's only few of these objects so the synchronization is going to be fast, but it is still better to avoid such usage in order to keep the startup process fast.